src/main/scala/com/tapad/docker/DockerComposePlugin.scala

```diff
@@ -154,6 +163,36 @@ class DockerComposePluginLocal extends AutoPlugin with ComposeFile with DockerCo
     }
   }
 
+  def restartRunningInstance(state: State, args: Seq[String]): State = {
+    try {
+      restartInstancePrecheck(state, args)
+      val newState = stopRunningInstances(state, args)
+      launchInstanceWithLatestChanges(newState, args)
+    } catch {
+      case ex: IllegalArgumentException =>
+        printError(ex.getMessage)
+        state
+      case ex: ComposeFileFormatException =>
+        printError(ex.getMessage)
+        state
+    }
+  }
+
+  def restartInstancePrecheck(state: State, args: Seq[String]): Unit = {
+    val newState = getPersistedState(state)
+    val runningInstanceIds = getServiceRunningInstanceIds(newState)
+
+    if (args.isEmpty) {
+      if (runningInstanceIds.size > 1)
+        throw new IllegalArgumentException("More than one running instance from the current sbt project was detected. " +
+          "Please provide an Instance Id parameter to the dockerComposeRestart command specifying which instance to stop.")
+    } else {
+      val restartList = runningInstanceIds.filter(instanceId => args.contains(instanceId))
```

Comment:
Can probably change this to:

`val restartList = runningInstanceIds.filter(args.contains(_))`